forms/serializers/form.py

```diff
@@ -330,34 +331,41 @@ def to_representation(self, instance):
             if check_for_none is None:
                 ret[field.field_name] = None
             elif field.label == "Entries data":
-                entries_dict = dict()
-                for entry in attribute.all():
-                    path_parts = entry.path.split(sep=".")
-                    value = True
-                    help_dict = dict()
-                    for part in reversed(path_parts):
-                        new_dict = {}
-                        if value:
-                            help_dict[part] = {
-                                "fields": {
-                                    entry.field.identifier: {
-                                        "value": entry.value,
-                                        "extra_value": entry.extra_value,
-                                    }
-                                }
-                            }
-                            value = False
-                        else:
-                            new_dict[part] = help_dict
-                            help_dict = new_dict
-                    entries_dict = always_merger.merge(entries_dict, help_dict)
-
-                ret[field.field_name] = entries_dict
+                ret[field.field_name] = self.create_entry(attribute)
             else:
                 ret[field.field_name] = field.to_representation(attribute)
 
         return ret
 
+    @staticmethod
+    def create_entry(attribute):
+        entries_dict = dict()
+        for entry in attribute.all():
+            path_parts = entry.path.split(sep=".")
+            try:
+                entry_value = literal_eval(entry.value)
+            except (SyntaxError, ValueError):
+                entry_value = entry.value
+            value = True
+            help_dict = dict()
+            for part in reversed(path_parts):
+                new_dict = {}
+                if value:
+                    help_dict[part] = {
+                        "fields": {
+                            entry.field.identifier: {
+                                "value": entry_value,
+                                "extra_value": entry.extra_value,
+                            }
+                        }
+                    }
+                    value = False
+                else:
+                    new_dict[part] = help_dict
+                    help_dict = new_dict
+            entries_dict = always_merger.merge(entries_dict, help_dict)
```

Comment:
I am pretty sure it is not going to work

Reply:
I am pretty sure it will :D You don't need to save the return value from always_merger.merge -function into entries_dict -variable. You can just return the value the function returns. But this is not a big deal anyways.

Reply:
or does the always_merger.merge even return any value? or does it just merge the help_dict into original entries_dict.

If this is the case, you don't need to set the entries_dict value on this row at all. Just call the function and then return the entries_dict.

Reply:
good catch

Reply:
I believe that if you would return always_merger.merge() then it would merge only last entry